### PR TITLE
add docs for irondb-relay and conflict_resolver

### PR DESCRIPTION
**Pull request by rileyberton**

### CODE REVIEW

File: configuration.md [hunk 1]
```diff
@@ -403,6 +404,33 @@ future, to accommodate clients with incorrect clocks.
 
 Default: 1 day
 
+#### raw_database conflict_resolver
+
+When a metric gets written more than one time at the same exact millisecond
```
Comment (esproul) [inline configuration.md]: "at the exact millisecond"

File: configuration.md [hunk 2]
```diff
@@ -403,6 +404,33 @@ future, to accommodate clients with incorrect clocks.
...
+offset you have a conflict we have to resolve. All operations in IRONdb are
+commutative and this lets us avoid complicated consensus algorithms for data.
+Conflicts, therefore, need to choose a winner and this choice needs to be
+consistent across the cluster. IRONdb gives you the following choices for
+conflict resolution should a datapoint appear more than once at the same
+millisecond.
+
+* `abs_biggest` - save the largest by absolute value.
+* `last_abs_biggest` - if used with the [IRONdb-relay](irondb-relay.md)
```
Comment (esproul) [inline configuration.md]: What happens if I set this but am not ingesting via irondb-relay?  Should try to be clear about whether these should be used at all without the relay.

File: irondb-relay.md [hunk 3]
```diff
@@ -0,0 +1,609 @@
...
+the [ZFS on Linux](https://github.com/zfsonlinux/zfs/wiki/Getting-Started)
+project. The setup script expects a zpool to exist, but you do not need to
+create any filesystems or directories ahead of time. Please refer to the
+appendix [ZFS Setup Guide](/zfs-guide.md) for details and examples.
+
+The following network protocols and ports are utilized. These are defaults and
+may be changed via configuration files.
+
+* 2003/tcp (Carbon plaintext submission)
+* 8112/tcp (admin UI, HTTP REST API)
```
Comment (esproul) [inline irondb-relay.md]: The relay listens on 8112?  Surprising to me, but if true, ignore this comment.

File: irondb-relay.md [hunk 4]
```diff
@@ -0,0 +1,609 @@
...
+
+### System Tuning
```
Comment (esproul) [inline irondb-relay.md]: Rather than duplicating system tuning, why not link back to the original?

File: irondb-relay.md [hunk 5]
```diff
@@ -0,0 +1,609 @@
...
+     *\(optional\)* Control enablement of durable delivery.  Default is "false".
+     If set to "true", will cause IRONdb-relay to use the disk to persist all 
+     incoming metrics to the file system before sending them on to IRONdb nodes.
+
+Run the setup script. All required options must be present, either as
+environment variables or via command-line arguments. A mix of environment
+variables and arguments is permitted, but environment variables take precedence
+over command-line arguments. Use the `-h` option to view a usage summary:
+
+    Usage: ./setup-irondb-relay [-h] -c <check-name> -u <check-uuid>
```
Comment (esproul) [inline irondb-relay.md]: The -B and -d options are missing from the one-line summary

File: irondb-relay.md [hunk 6]
```diff
@@ -0,0 +1,609 @@
...
+      </ruleset>
+     </filter>
+     ```
+     
+     The above would blacklist everything that starts `relay_test.agent.2` and
+     allow everything else.
+     
+     For best performance, it is wise to organize your `<rule>` blocks in
+     descending order based on the expected frequency of matching. You want the
+     `<rule>`s that match a lot to be at the beginning of the list and the
```
Comment (esproul) [inline irondb-relay.md]: would prefer "often" or "more often" to "a lot"

File: irondb-relay.md [hunk 7]
```diff
@@ -0,0 +1,609 @@
...
+<listener address="*" port="8112" backlog="100" type="http_rest_api">
+  <config>
+    <document_root>/opt/circonus/share/snowth-web</document_root>
+  </config>
+</listener>
+```
+
+The main listener serves multiple functions:
+* [HTTP REST API](api.md)
+* [Cluster replication](operations.md#replication) (TCP) and gossip (UDP)
```
Comment (esproul) [inline irondb-relay.md]: Does the relay actually do this?

File: irondb-relay.md [hunk 8]
```diff
@@ -0,0 +1,609 @@
...
+Default: 100
+
+##### Main listener type
+
+The type of libmtev listener this is. The main listener is configured to be
+only a REST API listener. This value should not be changed.
+
+Default: http_rest_api
+
+#### Graphite listener
```
Comment (esproul) [inline irondb-relay.md]: Are the graphite listeners supposed to match what's in IRONdb?  If so, maybe mention that they should be kept in sync?

File: configuration.md (hunk 2 above)
Comment (rileyberton) [reply configuration.md]: It's fine to set and not use the relay... generation counters would always be zero (unless/until we come up with a way to let people overwrite generation outside of relay usage)

File: irondb-relay.md (hunk 3 above)
Comment (rileyberton) [reply irondb-relay.md]: yes.  Has an Admin UI

File: irondb-relay.md (hunk 4 above)
Comment (rileyberton) [reply irondb-relay.md]: yes that's better

**Pull request merged**